cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -877,10 +877,10 @@ if(builtin_xrootd)
   set(XROOTD_DESTDIR ${CMAKE_BINARY_DIR})
   set(XROOTD_ROOTDIR ${XROOTD_DESTDIR})
   message(STATUS "Downloading and building XROOTD version ${xrootd_version}")
-  string(REPLACE "-Wall " "" __cxxflags "${ROOT_EXTERNAL_CXX_FLAGS}")  # Otherwise it produces many warnings
-  string(REPLACE "-W " "" __cxxflags "${__cxxflags}")          # Otherwise it produces many warnings
-  string(REPLACE "-Wshadow" "" __cxxflags "${__cxxflags}")          # Otherwise it produces many warnings
-  string(REPLACE "-Woverloaded-virtual" "" __cxxflags "${__cxxflags}")  # Otherwise it produces manywarnings
+
+  # Because XRootD will add the -Werror flag, we suppress all errors upfront
+  # to ensure forward compatibility with new compilers.
+  set(ROOT_XROOTD_CXX_FLAGS "${ROOT_EXTERNAL_CXX_FLAGS} -w")
```

Comment:
Isn't their `-Werror` added behind our `-w`? In that case and IIRC their `-Werror` wins...

Reply:
Good point. However, the current behaviour does not change the already existing logic. So apparently it works :)

Reply:
Godbolt says in case of `-w -Wall -Werror`, the `-w` wins :)

Reply:
[this so post](https://stackoverflow.com/questions/60157147/std-c14-after-std-c11-in-my-build-command-which-wins?noredirect=1&lq=1) and all the related agree that generally the rightmost option is the one picked by cmake, though there is no direct mention of the warning options

Reply:
It's the compiler picking options - CMake is just passing them. And my experiment with
```
$echo 'void f(int k) { }' | g++ -Wextra -Wall -Werror -w -fsyntax-only -x c++ -
```
tells me that the position of `-w` does not matter, and as @stwunsch put it: `-w` wins!